packages/proxy/lib/http/response-middleware.ts

```diff
@@ -393,99 +393,99 @@ const MaybePreventCaching: ResponseMiddleware = function () {
-  isLocalhost: boolean
-  url: URL
-  ctxDebug: Debug.Debugger
-}
-
-const cookieSameSiteRegex = /SameSite=(\w+)/i
-const cookieSecureRegex = /(^|\W)Secure(\W|$)/i
-const cookieSecureSemicolonRegex = /;\s*Secure/i
-
-const ensureSameSiteNone = ({ cookie, browser, isLocalhost, url, ctxDebug }: EnsureSameSiteNoneProps) => {
-  ctxDebug('original cookie: %s', cookie)
+const CopyCookiesFromIncomingRes: ResponseMiddleware = async function () {
+  const cookies: string | string[] | undefined = this.incomingRes.headers['set-cookie']
 
-  if (cookieSameSiteRegex.test(cookie)) {
-    ctxDebug('change cookie to SameSite=None')
-    cookie = cookie.replace(cookieSameSiteRegex, 'SameSite=None')
-  } else {
-    ctxDebug('add SameSite=None to cookie')
-    cookie += '; SameSite=None'
+  if (!cookies || !cookies.length) {
+    return this.next()
   }
 
-  const isFirefox = browser.family === 'firefox'
-
-  // Secure is required for SameSite=None cookies to be set in secure contexts
-  // (https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy),
-  // but will not allow the cookie to be set in an insecure context.
-  // Normally http://localhost is considered a secure context (see
-  // https://w3c.github.io/webappsec-secure-contexts/#localhost), but Firefox
-  // does not consider the Cypress-launched browser to be a secure context (see
-  // https://github.com/cypress-io/cypress/issues/18217). For that reason, we
-  // remove Secure from http://localhost cookies in Firefox.
-  if (cookieSecureRegex.test(cookie)) {
-    if (isFirefox && isLocalhost && url.protocol === 'http:') {
-      ctxDebug('remove Secure from cookie')
-      cookie = cookie.replace(cookieSecureSemicolonRegex, '')
+  // Cross-origin Cookie Handling
+  // ---------------------------
+  // - We capture cookies sent by responses and add them to our own server-side
+  //   tough-cookie cookie jar. All request cookies are captured, since any
+  //   future request could be cross-origin even if the response that sets them
+  //   is not.
+  // - If we sent the cookie header, it would fail to be set by the browser
+  //   (in most cases). We change the header name to 'X-Set-Cookie' to make it
+  //   clear that it's one we're handling ourselves.
+  // - We also set the cookies through automation so they are available in the
+  //   browser via document.cookie and via Cypress cookie APIs
+  //   (e.g. cy.getCookie). This is only done for cross-origin responses, since
+  //   non-cross-origin responses will be successfully set in the browser
+  //   automatically.
+  // - In the request middleware, we retrieve the cookies for a given URL
+  //   and attach them to the request, like the browser normally would.
+  //   tough-cookie handles retrieving the correct cookies based on domain,
+  //   path, etc. It also removes cookies from the cookie jar if they've expired.
+  const needsCrossOriginHandling = checkNeedsCrossOriginHandling(this)
+
+  const appendCookie = (cookie) => {
```

Comment:
```suggestion
  const appendCookie = (cookie: string) => {
```

Reply:
Addressed in [`95124a3` (#22320)](https://github.com/cypress-io/cypress/pull/22320/commits/95124a38297f4924a8a11dec21668171fa50d514)